distribution/lib/Standard/Base/0.0.0-dev/src/Data/Decimal.enso

```diff
@@ -354,7 +357,7 @@ type Decimal
              c = a + b
              # => Decimal.new 30.55
     + : Decimal -> Decimal ! Arithmetic_Error
-    + self (that : Decimal) -> Decimal ! Arithmetic_Error = self.add that
+    + self (that : Decimal) = self.add that
```

Comment:
Why are these return type check being removed? It seems suspicious.

Reply:
If I leave the checked annotations in, I get this:

```
        Reason: An unexpected panic was thrown: Type error: expected the result of `+` to be Decimal, but got <ERR>.
```

I'll be submitting a bug for this.

Reply:
Oh, weird. Okay then, I assume we should re-add once the bug is fixed.

Reply:
Added a TODO.

Reply:
I can reproduce the failure as
```
sbt:enso> runEngineDistribution --run test/Base_Tests Arithmetic.with.decimal.and.implicitly.converted.float.should.fail
```
correct? Then the problem is that there is a `WithWarning` instance over `DataflowError` instance. Do we really want to support attaching warnings to errors? It doesn't make sense to re-attach warnings to `DataflowError`. Possible fix:
```diff
diff --git engine/runtime/src/main/java/org/enso/interpreter/runtime/error/Warning.java engine/runtime/src/main/java/org/enso/interpreter/runtime/error/Warning.java
index e1ff24d35c..f750e377d6 100644
--- engine/runtime/src/main/java/org/enso/interpreter/runtime/error/Warning.java
+++ engine/runtime/src/main/java/org/enso/interpreter/runtime/error/Warning.java
@@ -73,7 +73,7 @@ public final class Warning implements EnsoObject {
       description = "Attaches the given warning to the value.",
       autoRegister = false)
   @Builtin.Specialize
-  public static WithWarnings attach(
+  public static EnsoObject attach(
       EnsoContext ctx, WithWarnings value, Object warning, Object origin) {
     return value.append(ctx, new Warning(warning, origin, ctx.nextSequenceId()));
   }
@@ -83,7 +83,7 @@ public final class Warning implements EnsoObject {
       description = "Attaches the given warning to the value.",
       autoRegister = false)
   @Builtin.Specialize(fallback = true)
-  public static WithWarnings attach(EnsoContext ctx, Object value, Object warning, Object origin) {
+  public static EnsoObject attach(EnsoContext ctx, Object value, Object warning, Object origin) {
     return WithWarnings.wrap(ctx, value, new Warning(warning, origin, ctx.nextSequenceId()));
   }
 
diff --git engine/runtime/src/main/java/org/enso/interpreter/runtime/error/WithWarnings.java engine/runtime/src/main/java/org/enso/interpreter/runtime/error/WithWarnings.java
index 2968eaf419..0bb3191157 100644
--- engine/runtime/src/main/java/org/enso/interpreter/runtime/error/WithWarnings.java
+++ engine/runtime/src/main/java/org/enso/interpreter/runtime/error/WithWarnings.java
@@ -120,9 +120,11 @@ public final class WithWarnings implements EnsoObject {
     return goodValue;
   }
 
-  public static WithWarnings wrap(EnsoContext ctx, Object value, Warning... warnings) {
+  public static EnsoObject wrap(EnsoContext ctx, Object value, Warning... warnings) {
     if (value instanceof WithWarnings with) {
       return with.append(ctx, warnings);
+    } else if (value instanceof DataflowError err) {
+      throw err;
     } else {
       return new WithWarnings(value, ctx.getWarningsLimit(), warnings);
     }
```
CCing @Akirathan.

Reply:
This patch is resulting in this:

```
Execution finished with an error: Wrapped_Dataflow_Error.Error
        at <enso> Panic.throw(Internal)
        at <enso> Panic.type.rethrow_wrapped_if_error<arg-1>(Panic.enso:223:32-85)
        at <enso> Panic.type.rethrow_wrapped_if_error(Panic.enso:223:9-96)
        at <enso> Builder.append_vector_range(Vector.enso:1409-1415)
        at <enso> case_branch(Suite.enso:99:25-68)
        at <enso> Suite.run_with_filter.junit_sb_builder(Suite.enso:95-101)
        at <enso> Array_Like_Helpers.each.Array_Like_Helpers.each(Array_Like_Helpers.enso:301:9-24)
        at <enso> Range.each.go<arg-1>(Range.enso:262:36-51)
        at <enso> Range.each.go<arg-2>(Range.enso:262:21-52)
        at <enso> Range.each.go(Range.enso:261-263)
        at <enso> Test_Reporter.wrap_junit_testsuites(Test_Reporter.enso:26:14-19)
        at <enso> Suite.run_with_filter(Suite.enso:91-101)
        at <enso> Decimal_Spec.main(Decimal_Spec.enso:1068:5-32)
```